content/documentation/pages/5-feature-guides/1-general/1-applicaiton-metadata.md

```diff
@@ -0,0 +1,280 @@
+---
+path: 'feature-guides/general/application-metadata/'
+title: 'Application Metadata'
+description: 'Create and use application properties metadata'
+---
+
+## Application Metadata
+
+Stream and Task applications are Spring Boot applications that are aware of many [Common Application Properties](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#spring-cloud-dataflow-global-properties), such as server.port but also families of properties such as those with the prefix spring.jmx and logging. For this you need to enable the [Configuration Metadata](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-configuration-metadata.html) generation and to specify the whitelisted properties.
+
+Add the `spring-boot-configuration-processor` as an optional dependency to generate configuration metadata file for your application properties:
+
+```xml
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-configuration-processor</artifactId>
+    <optional>true</optional>
+</dependency>
+```
+
+The the configuration metadata files will be located under `META-INF/spring-configuration-metadata.json`.
+
+In addition to the generated configuration metadata, when creating your own application, you should `whitelist` the primary properties so that the shell and the UI can display them first when presenting options through TAB completion or in drop-down boxes.
+
+To whitelist application properties, create a file named `spring-configuration-metadata-whitelist.properties` under the `META-INF` resource directory. There are two property keys that can be used inside this file. The first key is named `configuration-properties.classes`. The value is a comma separated list of fully qualified `@ConfigurationProperty` class names. The second key is `configuration-properties.names`, whose value is a comma-separated list of property names. This can contain the full name of the property, such as `server.port`, or a partial name to whitelist a category of property names, such as `spring.jmx`.
+
+The [Spring Cloud Stream application starters](https://github.com/spring-cloud-stream-app-starters) are a good place to look for examples of usage. The following example comes from the file sink’s spring-configuration-metadata-whitelist.properties file:
+
+```properties
+configuration-properties.classes=org.springframework.cloud.stream.app.file.sink.FileSinkProperties
+```
+
+If we also want to add `server.port` to be white listed, it would become the following line:
+
+```properties
+configuration-properties.classes=org.springframework.cloud.stream.app.file.sink.FileSinkProperties
+configuration-properties.names=server.port
+```
+
+The benefits of a companion artifact include:
```

Comment:
No mention of what a companion artifact is...  The general idea of how to package them up - if you are using .jars, can be inside the application jar (pros/cons) or as a 'companion metadata jar'  (pros/con) and also how it is done for docker isn't a 'companion artifact' approach... so the term really only applies for jar packaging.